cndict/bundle_friso.py

```diff
@@ -0,0 +1,307 @@
+        self._fp.write(struct.pack("!I", self.VERSION))
+        self._fp.flush()
+        self.compressed_size = 0
+        self.full_size = 4 # For the 'version' byte
+
+    def _write_data(self, data):
+        self._fp.write(data)
+        self.compressed_size += len(data)
+
+    def flush(self, is_final=False):
+        if not self._use_compression:
+            self._write_data(self._buf)
+        else:
+            # Flush any outstanding data in the buffer
+            self._write_data(self._compressor.compress(bytes(self._buf)))
+
+            if is_final:
+                self._write_data(self._compressor.flush(zlib.Z_FINISH))
+
+        self._fp.flush()
+        self.full_size += len(self._buf)
+        self._buf = bytearray()
+
+    def _maybe_flush(self):
+        if len(self._buf) > self.CHUNK_SIZE:
+            self.flush()
+
+    def add_entry(self, lextype, term, syns, freq):
+        # Perform the encoding...
+        header = LEXTYPE_MAP[lextype]
+
+        if syns:
+            header |= F_SYNS
+        if freq:
+            header |= F_FREQS
+
+        self._buf.append(header)
+        self._buf += term
+        self._buf.append(0) # NUL terminator
+
+        if syns:
+            self._buf += struct.pack("!h", len(syns))
+            for syn in syns:
+                self._buf += syn
+                self._buf.append(0)
+        if freq:
+            self._buf += struct.pack("!I", freq)
+
+        self._maybe_flush()
+
+
+def process_lex_entry(type, file, buf):
+    print type, file
+    fp = open(file, 'r')
+    for line in fp:
+        line = line.strip()
+        comps = line.split('/')
+        # print comps
+        term = comps[0]
+        syns = comps[1].split(',') if len(comps) > 1 else []
+        if len(syns) == 1 and syns[0].lower() == 'null':
+            syns = []
+        freq = int(comps[2]) if len(comps) > 2 else 0
+
+        buf.add_entry(type, term, syns, freq)
+        # print "Term:", term, "Syns:", syns, "Freq", freq
+        # Now dump it, somehow
+
+
+def strip_comment_lines(blob):
+    lines = [line.strip() for line in blob.split('\n')]
+    lines = [line for line in lines if line and not line.startswith('#')]
+    return lines
+
+
+def sanitize_file_entry(typestr, filestr):
+    typestr = strip_comment_lines(typestr)[0]
+    filestr = strip_comment_lines(filestr)
+    filestr = [f.rstrip(';') for f in filestr]
+    return typestr, filestr
+
+
+lexre = re.compile(r'([^:]+)\w*:\w*\[([^\]]*)\]', re.MULTILINE)
+lexindex = os.path.join(lexdir, 'friso.lex.ini')
+lexinfo = open(lexindex, 'r').read()
+matches = lexre.findall(lexinfo)
+# print matches
+
+dstdir = opts.out
+if opts.mode == 'c':
+    dstfile = 'cndict_data.c'
+else:
+    dstfile = 'cndict_data.out'
+
+try:
+    os.makedirs(dstdir)
+except OSError as e:
+    if e.errno != errno.EEXIST:
+        raise
+
+dstfile = os.path.join(dstdir, dstfile)
+ofp = open(dstfile, 'w')
+
+if opts.mode == 'c':
+    ofp.write(r'''
+// Generated by python script
```

Comment:
Be a bit more verbose, what's this file, and why was it generated by a python script, which python script etc...